### ISSUE TITLE
Integrate Nitrokey support to let users sign, encrypt and decrypt files

### ISSUE THREAD

**Issue by 4jNsY6fCVqZv** [elementary/files#862]
What it says on the tin!
This issue is thematically well suited to the following discussions: elementary/pantheon-agent-polkit#33 and elementary/pantheon-agent-polkit#40 and elementary/greeter#230 and elementary/mail#345

Unlike a Yubikey, it is Free Hardware and Free Software, which is mostly installed locally (in Berlin, Germany).

Here you can find more information: https://www.nitrokey.com/ & https://github.com/nitrokey

This integration can also be used to decrypt your hard disks - see LUKS/LUKS2 - or as a solution for Two-Factor-Authentication in the Installer and Switchboard Online Accounts Plug. Integration with Mail (to sign, encrypt and decrypt emails) and for installing Software with AppCenter it's also very useful.

**Comment by 4jNsY6fCVqZv** [comment]
@jeremeypw What do you think about support for actions with a hardware based token in Files? Of course this is also exciting as a cross-system thought as you can see from the other linked issues. But when you first think about the chances and possibilities for Files ...

**Comment by jeremypw** [comment]
Files should certainly support any system-wide hardware encryption facility.  Not sure whether it needs to be, or should be, tied to a particular device.  Adding the ability to encrypt/decrypt files easily would be nice if it can be done with a contractor or plugin.

**Comment by 4jNsY6fCVqZv** [comment]
Do you know if the team is already discussing ideas to enable system-wide encryption using hardware tokens? For example, Purism has introduced a way for its system https://puri.sm/products/librem-key/. Basically, that was a branded Nitrokey before they started their own production. Of course the support should not be tied to a particular device. But Nitrokeys are a good reference because they are based on Free Hardware and Free Software.

**Comment by jeremypw** [comment]
There is a #security channel in the elementary Slack workspace but there does not appear to have been any recent activity. I could post a link to this issue there.

**Comment by 4jNsY6fCVqZv** [comment]
yes, that would be wonderful and supportive. thank you very much!

**Comment by jeremypw** [comment]
I posted an invitation to comment on this issue in the elementary Slack #security channel (16 memebers) a couple of weeks ago but so far no response.

**Comment by 4jNsY6fCVqZv** [comment]
Oh, thank you, do you have any idea why? And do you know if Daniel or Cassidy already have thoughts on this subject?

**Comment by jeremypw** [comment]
@4jNsY6fCVqZv That channel seems to be little used it seems.  I have reposted in the general channel.

**Comment by 4jNsY6fCVqZv** [comment]
@jeremypw Thank you, I'm curious! Do you know how far the topic of authentication is already being discussed? There are also some mockups that go in that direction. https://github.com/elementary/pantheon-agent-polkit/issues/33#issuecomment-417801703

**Comment by jeremypw** [comment]
@4jNsY6fCVqZv I haven't seen anything recent in Slack although I can only search the last 10,000 messages. That polkit issue is interesting though it seems to have stalled. Its not really my area of expertise unfortunately.

**Comment by 4jNsY6fCVqZv** [comment]
@jeremypw Thank you for investigating! Is there a direct contact person for such an issue?

**Comment by jeremypw** [comment]
I am not sure who the current maintainer of polkit-agent is but @donadigo  wrote a lot of the code...

**Comment by jeremypw** [comment]
The creation of transparently encrypted disks, partitions and folders is outside the scope of Files. This is best done at system level I think.  Files would be best used for creating an encrypted version of a file or set of files suitable for e.g. sending by email; and for reversing such a process.

**Comment by jeremypw** [comment]
I have created a simple contractor for encrypting and decrypting individual files here: https://github.com/jeremypw/gpg_wrapper.  As it is an initial release, it may not yet have all the functionality desired.

**Comment by 4jNsY6fCVqZv** [comment]
Thanks for your work on this!
I think it would be okay for me to close this issue if your tool works well and it is clear that Files doesn't want to offer this functionality out of the box.

**Comment by jeremypw** [comment]
You might as well leave it open until someone flags it as "Out of scope"
